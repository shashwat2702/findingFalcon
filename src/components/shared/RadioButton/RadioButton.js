import React from 'react';
import PropTypes from 'prop-types';
import './RadioButton.scss';

const RadioButton = ({ vehicles }) => (
  <div>
    {
          vehicles.map(vehicle => (
            <div key={vehicle.name} className="dropdown">
              <input type="radio" name="gender" value={vehicle.name} disabled={false} />
              {' '}
              {vehicle.name}
              {` (${vehicle.total_no})`}
              <br />
            </div>
          ))
      }
  </div>
);

RadioButton.propTypes = {
  // eslint-disable-next-line react/forbid-prop-types
  vehicles: PropTypes.array.isRequired,
};
export default RadioButton;
