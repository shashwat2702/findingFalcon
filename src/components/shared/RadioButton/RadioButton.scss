.dropdown {
    color: white;
    font-size: 1.2rem;
    padding-top: 1rem;
}