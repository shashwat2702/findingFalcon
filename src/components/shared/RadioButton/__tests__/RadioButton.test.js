import React from 'react';
import renderer from 'react-test-renderer';
import RadioButton from '../RadioButton';


jest.mock('react-router-dom');

const onRadioClickMocked = jest.fn();
const planets = [{ name: 'PlanetA', distance: 100 }, { name: 'PlanetB', distance: 150 },
  { name: 'PlanetC', distance: 200 }, { name: 'PlanetD', distance: 50 },
  { name: 'PlanetE', distance: 250 }, { name: 'PlanetF', distance: 300 }];
const vehicles = [{ name: 'VehicleA', speed: 10 }, { name: 'VehicleB', speed: 5 },
  { name: 'VehicleC', speed: 20 }, { name: 'VehicleD', speed: 25 },
  { name: 'VehicleE', speed: 50 }, { name: 'VehicleF', speed: 40 }];
const selectedPlanets = ['PlanetC', 'PlanetF', 'PlanetD', 'PlanetA'];


describe('RadioButton snapshot', () => {
  it('should render correctly', () => {
    const wrapper = renderer.create(<RadioButton
      vehicles={vehicles}
      selectedPlanets={selectedPlanets}
      planets={planets}
      onRadioClick={onRadioClickMocked}
      planetNumber={1}
      usedVehicles={{ name: 'VehicleC', speed: 20 }}
    />).toJSON();
    expect(wrapper).toMatchSnapshot();
  });
});
