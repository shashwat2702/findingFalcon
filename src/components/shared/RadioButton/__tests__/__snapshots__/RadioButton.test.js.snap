// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioButton snapshot should render correctly 1`] = `
<div>
  <div
    className="dropdown"
  >
    <input
      disabled={true}
      name="vehicle1"
      onClick={[MockFunction]}
      type="radio"
      value="VehicleA"
    />
     
    VehicleA
     (NaN)
    <br />
  </div>
  <div
    className="dropdown"
  >
    <input
      disabled={true}
      name="vehicle1"
      onClick={[MockFunction]}
      type="radio"
      value="VehicleB"
    />
     
    VehicleB
     (NaN)
    <br />
  </div>
  <div
    className="dropdown"
  >
    <input
      disabled={true}
      name="vehicle1"
      onClick={[MockFunction]}
      type="radio"
      value="VehicleC"
    />
     
    VehicleC
     (NaN)
    <br />
  </div>
  <div
    className="dropdown"
  >
    <input
      disabled={true}
      name="vehicle1"
      onClick={[MockFunction]}
      type="radio"
      value="VehicleD"
    />
     
    VehicleD
     (NaN)
    <br />
  </div>
  <div
    className="dropdown"
  >
    <input
      disabled={true}
      name="vehicle1"
      onClick={[MockFunction]}
      type="radio"
      value="VehicleE"
    />
     
    VehicleE
     (NaN)
    <br />
  </div>
  <div
    className="dropdown"
  >
    <input
      disabled={true}
      name="vehicle1"
      onClick={[MockFunction]}
      type="radio"
      value="VehicleF"
    />
     
    VehicleF
     (NaN)
    <br />
  </div>
</div>
`;
